### ISSUE TITLE
ThreadLocalActiveSpan.Continuation.activate() overuse should error

### ISSUE THREAD

**Issue by tylerbenson** [opentracing/opentracing-java#165]
The [ActiveSpan javadoc](https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/ActiveSpan.java#L101) states that:
`It is an error to call activate() more than once on a single Continuation instance`

I propose that this should be asserted by `ThreadLocalActiveSpan` by returning a runtime exception if used more than once.  [Currently](https://github.com/opentracing/opentracing-java/blob/master/opentracing-util/src/main/java/io/opentracing/util/ThreadLocalActiveSpan.java#L145) it only returns a new `ThreadLocalActiveSpan`.

**Comment by rbtcollins** [comment]
Indeed please. I found this looking for bugs about deactivates silent refcount leaking when the API has been misused. While it may only be a test-support class, thats even more reason to be super strict about the API, to help folk use it right.

**Comment by pavolloffay** [comment]
Note that continuation API with refcounting will be probably replaced by Scopes https://github.com/opentracing/opentracing-java/issues/189

**Comment by tedsuo** [comment]
@tylerbenson can we close this since we are deprecating v0.30?

**Comment by carlosalberto** [comment]
So since we are not using this in the main api, but we kept the `Continuation` concept through the `io.opentracing.util.AutoFinishScope`, maybe we should create a PR against it?

Now, I don't know if throwing an `Exception` is a good idea for this case. Thoughts? @pavolloffay @tedsuo @tylerbenson ?

**Comment by tylerbenson** [comment]
I agree throwing an exception might be a bit draconian, but what would be the better behavior?

**Comment by carlosalberto** [comment]
@tylerbenson As of now, I think we cannot do much more, other than throwing an `Exception` - once we get some kind of Observer API we could simply show some kind of warning. So I think we *could* indeed throwing a `RuntimeException` for now. Would that work?

Again, this would be for the extra `AutoFinishScopeManager`, not the default `ThreadLocalScopeManager` one ;)
